src/neo/Ledger/Blockchain.cs

```diff
@@ -427,10 +427,18 @@ private void Persist(Block block)
                     clonedSnapshot.Transactions.Add(tx.Hash, state);
                     clonedSnapshot.Transactions.Commit();
 
+                    // For now, only support a first trace debug plugin that returns true to ShouldTrace
+                    var traceDebugPlugin = Plugin.TraceDebugPlugins
```

Comment:
Can we cache this_

Reply:
no, traceDebugPlugin may be different for each transaction.